compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/AbstractNodeFactory.java

```diff
@@ -189,17 +189,17 @@ public static Minutiae createEndOfLineMinutiae(String text) {
                         .collect(Collectors.toList())).createUnlinkedFacade());
     }
 
-    public static <T extends Node> NodeList<T> createSeparatedNodeList(T... nodes) {
+    public static <T extends Node> SeparatedNodeList<T> createSeparatedNodeList(Node... nodes) {
```

Comment:
Do we need to change the parameter type from the generic type `T` to `Node`?

Reply:
Yes, I think so.
The `SeparatedNodeList` contains separator nodes as well.
These are not of the Type T.

See issue #26440

Reply:
I have added a comment in 26440

Reply:
What if `SeparatedNodeList` was changed to accept the `Node` type, but does a runtime check to see if every even index is of Type `T`?

Reply:
oh okay, I think I misunderstood the problem.